File: EmployeePayrollWeb/src/app/classes/leave.ts
export interface ILeave {
    empName: string
    empContactNo: string
    employeeId: number
    leaveDate: string
    leaveId: number
    leaveReason: string
    noOfFullDayLeaves: number
    noOfHalfDayLeaves: number
}
export class leave {
    leaveId: number = 0;
    employeeId: number = 0;
    leaveDate?: Date = undefined;
    leaveReason: string = "";
    noOfFullDayLeaves?: number = 0;
    noOfHalfDayLeaves?: number = 0;
  }

File: EmployeePayroll.API/Models/Leave.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace EmployeePayroll.API.Models
{
    [Table("Leave")]
    public class Leave
    {
        [Key]
        public int LeaveId { get; set; }
        public int? EmployeeId { get; set; }
        public DateTime? LeaveDate { get; set; }
        public string? LeaveReason { get; set; }
        public int? NoOfFullDayLeaves { get; set; }
        public int? NoOfHalfDayLeaves { get; set; }
    }
}
